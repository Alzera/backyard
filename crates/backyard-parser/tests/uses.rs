use backyard_parser::parse_eval;

#[test]
fn basic() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "use const A\\B;").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn alias() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "use const A\\B as A;").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn multiple() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(&arena, "use A\\B as A, B\\C as B;").unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}

#[test]
fn items() {
  let arena = bumpalo::Bump::new();
  let asts = parse_eval(
    &arena,
    "use App\\Models\\{
  const User\\UserTesting as UserTestingA,
  User\\UserTestingB as UserTestingB,
  function UserTestingC
};"
  ).unwrap();
  insta::assert_yaml_snapshot!(asts.serializable());
}
