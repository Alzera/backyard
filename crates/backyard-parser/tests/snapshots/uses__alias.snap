---
source: crates/backyard-parser/tests/uses.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: use
    name: ~
    items:
      - node_type: use_item
        modifier: Const
        name:
          - 65
          - 92
          - 66
        alias:
          node_type: identifier
          name:
            - 65
          loc:
            start:
              line: 1
              column: 17
              offset: 17
            end:
              line: 1
              column: 18
              offset: 18
          leadings: ~
          trailings: ~
        loc:
          start:
            line: 1
            column: 4
            offset: 4
          end:
            line: 1
            column: 17
            offset: 17
        leadings: ~
        trailings: ~
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 1
        column: 18
        offset: 18
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 1
    column: 18
    offset: 18
leadings: ~
trailings: ~
